packages/lodestar/src/eth1/impl/ethers.ts

```diff
@@ -70,40 +70,63 @@ export class EthersEth1Notifier extends (EventEmitter as { new(): Eth1EventEmitt
       this.logger.verbose("Eth1 notifier is disabled" );
       return;
     }
-    if (this.started) {
-      this.logger.verbose("Eth1 notifier already started" );
+    await this.doStart();
```

Comment:
It's a bit hard to use eth1notifier now.
We could do following:
- remove starting from "node.ts" file as it makes no sense to be started there
- keep counter how much time it's started (every time some module calls start +1, and everytime module calls stop -1 and stop only we counter is gonna be zero after stop)

Reply:
I think we can (eventually) decompose this `IEth1Notifier` into a few parts, that can be run on-demand, rather than started globally. If we want to reuse resources, we could keep the provider and keep that around (it could be started/stopped in node.ts). Decomposing the functionality will lead to increased maintainability (each thing can be in its own file) and flexibility in how we can use this:
```typescript
type BlockTag = string | number

/**
 * Thin wrapper around ethers provider, tailored for the deposit contract
 */
interface IEth1Provider {
  /**
   * initialize contract, error if it doesn't exist or is invalid
   */
  start(): Promise<void>;
  stop(): Promise<void>;

  getBlock(blockTag: BlockTag): Promise<Block>;
  getDepositEvents(fromBlockTag: BlockTag, toBlockTag?: BlockTag): Promise<IDepositEvent[]>;

  // event for new blocks
  on(event: 'block', block: Block): void;
  off(event: 'block', block: Block): void;

  // event for module stopping, allows for event consumers to gracefully shut down
  on(event: 'stop'): void;
  off(event: 'stop'): void;
}

interface IBatchDepositEvents {
  depositEvents: IDepositEvent[];
  blockNumber: number;
}

/**
 * Stream of batched deposit events
 */
function getHistoricalDepositEventsStream(provider: IEth1Provider, from: blockNumber, to: blockNumber): Promise<Pushable<IBatchDepositEvents>>;

/**
 * Abortable stream of batched deposit events
 * data starting at `from`, continues at ETH1_FOLLOW_DISTANCE until aborted or IEthProvider stops
 */
function getFollowDepositEventsStream(provider: IEth1Provider, from: blockNumber): Promise<Pushable<IBatchDepositEvents>>;
```

From there we've separated how the data is consumed, database storage concerns, whether a stream has a single or multiple consumers, etc. We should write up a design doc determining what we store, when we store, which modules manage storage, etc. Likely, we will have a strategy for genesis and a strategy for proposing.

Eth1Data processing requires a depositData tree, either fetched from the db module, or managed by the genesis builder.
```typescript
/**
 * Create an Eth1Data based on a block and a deposit data root tree
 */
function getEth1Data(block: Block, depositDataRootTree: TreeBacked<Root<DepositData>[]>): {timestamp: number, eth1Data: Eth1Data};
```

Not saying we should do all this in this PR, just brainstorming how to make this whole process simpler and more manageable. IMO building out good primitives like this will help everything else fall into place.